src/tasks/deployment-of-bridged-token-deployer.ts

```diff
@@ -0,0 +1,148 @@
+
+import { defaultTokens, nativeTokenPriceGnosisChain } from "./ts/constants";
+import { defaultSafeDeploymentAddresses } from "./ts/safe";
+
+export const OUTPUT_FOLDER_GC = "./output/deployment-gc";
+
+const setupBridgedTokenDeployerTask: () => void = () => {
+  task(
+    "deployment-bridged-token-deployer",
+    "Generate the list of claims from a csv and reads settings from json and deploys the bridged token deployer on gnosis chain",
+  )
+    .addParam(
+      "claims",
+      "Path to the CSV file that contains the list of claims to generate.",
+    )
+    .addParam(
+      "settings",
+      "Path to the JSON file that contains the deployment settings.",
+    )
+    .setAction(generateDeployment);
+};
+
+async function generateDeployment(
+  { claims: claimCsv, settings: settingsJson }: Args,
+  hre: HardhatRuntimeEnvironment,
+): Promise<void> {
+  const { ethers } = hre;
+  const [deployer] = await ethers.getSigners();
+
+  const chainId = (await hre.ethers.provider.getNetwork()).chainId.toString();
+  if (chainId !== "100") {
+    throw new Error(
+      `This script must be run on gnosis chain. Found chainId ${chainId}`,
+    );
+  }
+
+  const inputSettings: Settings = JSON.parse(
+    await fs.readFile(settingsJson, "utf8"),
+  );
+  console.log(`Using deployer ${deployer.address}`);
+
+  console.log("Reading user claims for gnosis chain from file...");
+  const claims = await parseCsvFile(claimCsv);
+
+  console.log("Generating Merkle proofs...");
+  const { merkleRoot, claims: claimsWithProof } = computeProofs(claims);
+
+  const settings = {
+    ...inputSettings,
+    virtualCowToken: dummyVirtualTokenCreationSettings,
+    multiTokenMediatorGnosisChain: constants.AddressZero,
+  };
+
+  // In the following function, we are generating the addresses, as they would
+  // be generated within the mainnet deployment script - but with many zero
+  // addresses and hashes, as displayed in the settings definition
+  // Hence, we rely on the fact that the addresses computed by generateProposal
+  // for the cowDao and cowToken do not depend on the virtual token deployment
+  // parameters.
+  // For double security, one can also provide the expected values in as expected
+  // setting variables
+  const {
+    addresses: { cowToken, cowDao },
+  } = await generateProposal(
+    settings,
+    defaultSafeDeploymentAddresses(chainId),
+    hre.ethers,
+  );
+  if (settings.cowToken.expectedAddress !== cowToken) {
+    if (settings.cowToken.expectedAddress !== undefined) {
+      throw new Error("Expected cowToken address must be defined");
+    } else {
+      console.warn("settings.cowToken.expectedAddress was not defined");
+    }
+  }
+
+  if (settings.cowDao.expectedAddress !== cowDao) {
+    if (settings.cowDao.expectedAddress !== undefined) {
+      throw new Error("Expected cowDao address must be defined");
+    } else {
+      console.warn("settings.cowToken.expectedAddress was not defined");
+    }
+  }
```

Comment:
I think this should be done inside the proposal generation and fail silently if unset, so it can be tested. Anyway this is extremely minor so let's merge anyway!